src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -36,6 +39,7 @@ public Socket(EventBus eventBus, SocketHint<T> socketHint, T value) {
         this.eventBus = eventBus;
         this.socketHint = socketHint;
         this.value = value;
+        this.direction = Direction.INPUT;
```

Comment:
How can I force that? Throw an exception if it's not specified? Keep in mind that this field is set by steps, not operaions.  Operations specify which sockets are inputs buy return them from a certain method, and Steps immediately tag them with an enum (since that enum really describes the socket's relationship with the step,  and also for practical reasons already outlined)